### ISSUE TITLE
Incorrect broadcasted Period subtraction behaviour?

### ISSUE THREAD

**Issue by GordStephen** [JuliaLang/julia#12094]
The behaviour for broadcasted `Period` subtraction is currently
```julia
julia> Day(1) - [Day(2), Day(3)]
2-element Array{Base.Dates.Day,1}:
 1 day 
 2 days
```
and
```julia
julia> Day(1) .- [Day(2), Day(3)]
2-element Array{Base.Dates.Day,1}:
 1 day 
 2 days
```
whereas presumably the desired output is:
```julia
julia> [Day(1) - Day(2), Day(1) - Day(3)]
2-element Array{Base.Dates.Day,1}:
 -1 day 
 -2 days
```

Looking at [dates/periods.jl](https://github.com/JuliaLang/julia/blame/master/base/dates/periods.jl#L45-52), the first two method definitions here work fine for `+` but not `-`:
```julia
for op in (:.+, :.-)
op_ = symbol(string(op)[2:end])
    @eval begin
        ($op){P<:Period}(x::P,Y::StridedArray{P}) = ($op)(Y,x)
        ($op_){P<:Period}(x::P,Y::StridedArray{P}) = ($op)(Y,x)
        ($op_){P<:Period}(Y::StridedArray{P},x::P) = ($op)(Y,x)
    end
end
```

One option might be to implement unary addition (as identity) and then do
```julia
for op in (:.+, :.-)
op_ = symbol(string(op)[2:end])
    @eval begin
        ($op){P<:Period}(x::P,Y::StridedArray{P}) = ($op_)(($op)(Y,x))
        ($op_){P<:Period}(x::P,Y::StridedArray{P}) = ($op_)(($op)(Y,x))
        ($op_){P<:Period}(Y::StridedArray{P},x::P) = ($op)(Y,x)
    end
end
```
or, just hardcode the definitions as 
```julia
(.+){P<:Period}(x::P,Y::StridedArray{P}) = Y .+ x
(+){P<:Period}(x::P,Y::StridedArray{P}) = Y .+ x
(+){P<:Period}(Y::StridedArray{P},x::P) = Y .+ x
(.-){P<:Period}(x::P,Y::StridedArray{P}) = (-Y) .+ x
(-){P<:Period}(x::P,Y::StridedArray{P}) = (-Y) .+ x
(-){P<:Period}(Y::StridedArray{P},x::P) = Y .- x
```
which would actually be fewer LOC... maybe more readable too.

**Comment by GordStephen** [comment]
I'm working on a PR to get addition/subtraction working with heterogeneous `Period` arrays right now, so I'm happy to fix this while I'm at it - is there a preference between the two approaches mentioned above?

**Comment by quinnj** [comment]
Closed via #12115
